src/main/java/org/icgc/argo/program_service/services/ProgramService.java

```diff
@@ -295,6 +290,29 @@ private void processRegions(@NonNull ProgramEntity programToUpdate, @NonNull Lis
     programRegionRepository.saveAll(programRegions);
   }
 
+  /**
+   * Compares the user provided collection against the colleciton stored in the system.
+   * Throws an exception if it determines that the users provided a collection that cannot
+   * be found by the system.
+   * @param errorTemplate String template for the exception message.
+   * @param userList The user provided collection.
+   * @param systemList The system collection.
+   */
+  @SuppressWarnings("unchecked")
```

Comment:
I don't think this code is quite right.

This test that I wrote fails; I think it should pass instead.

```
 @Test
  public void testCompareListsBadInput2() {
    val userList = List.of("baz");
    val systemList = List.of("bar", "foo");

    try {
      ProgramService.compareLists("%s", userList, systemList);
      Assert.fail("Shouldn't succeed");
    } catch (Exception e) {
      Assert.assertEquals(e.getMessage(), "INVALID_ARGUMENT:  baz");
    }
  }

```

Reply:
there is a precondition and implication that the user supplied list was used to retrieve the system list. The test you are providing here is for undefined behaviour. I'll update the Java doc to include that.